pyintegration/deephaven2/agg.py

```diff
@@ -16,257 +20,288 @@ class Aggregation:
+
+            return self._j_agg_spec.aggregation(*[_JPair.parse(col) for col in self._cols])
+        else:
+            raise DHError("No columns specified for the aggregation operation.")
 
+    @property
+    def j_agg_spec(self):
+        return self._j_agg_spec
 
-def sum_(cols: List[str]) -> Aggregation:
+
+def sum_(cols: List[str] = None) -> Aggregation:
     """ Create a Sum aggregation.
 
     Args:
-        cols (List[str]): the columns to aggregate on, can be renaming expressions, i.e. "new_col = col"
+        cols (List[str]): the columns to aggregate on, can be renaming expressions, i.e. "new_col = col"; default is
+            None, only valid when used in Table agg_all_by operation
 
     Returns:
         an aggregation
     """
-    return Aggregation(j_aggregation=_JAggregation.AggSum(*cols))
+    # return Aggregation(j_aggregation=_JAggregation.AggSum(*cols))
+    return Aggregation(j_agg_spec=_JAggSpec.sum(), cols=cols)
 
 
-def abs_sum(cols: List[str]) -> Aggregation:
+def abs_sum(cols: List[str] = None) -> Aggregation:
     """ Create an Absolute-sum aggregation.
 
     Args:
-        cols (List[str]): the columns to aggregate on, can be renaming expressions, i.e. "new_col = col"
+        cols (List[str]): the columns to aggregate on, can be renaming expressions, i.e. "new_col = col"; default is
+            None, only valid when used in Table agg_all_by operation
 
     Returns:
         an aggregation
     """
-    return Aggregation(j_aggregation=_JAggregation.AggAbsSum(*cols))
+    return Aggregation(j_agg_spec=_JAggSpec.absSum(), cols=cols)
 
 
-def group(cols: List[str]) -> Aggregation:
+def group(cols: List[str] = None) -> Aggregation:
     """ Create a Group aggregation.
 
     Args:
-        cols (List[str]): the columns to aggregate on, can be renaming expressions, i.e. "new_col = col"
+        cols (List[str]): the columns to aggregate on, can be renaming expressions, i.e. "new_col = col"; default is
+            None, only valid when used in Table agg_all_by operation
 
     Returns:
         an aggregation
     """
-    return Aggregation(j_aggregation=_JAggregation.AggGroup(*cols))
+    return Aggregation(j_agg_spec=_JAggSpec.group(), cols=cols)
 
 
-def avg(cols: List[str]) -> Aggregation:
+def avg(cols: List[str] = None) -> Aggregation:
     """ Create an Average aggregation.
 
     Args:
-        cols (List[str]): the columns to aggregate on, can be renaming expressions, i.e. "new_col = col"
+        cols (List[str]): the columns to aggregate on, can be renaming expressions, i.e. "new_col = col"; default is
+            None, only valid when used in Table agg_all_by operation
 
     Returns:
         an aggregation
     """
-    return Aggregation(j_aggregation=_JAggregation.AggAvg(*cols))
+    return Aggregation(j_agg_spec=_JAggSpec.avg(), cols=cols)
 
 
 def count_(col: str) -> Aggregation:
-    """ Create a Count aggregation.
+    """ Create a Count aggregation. It can not be used in Table agg_all_by().
```

Comment:
1) Other docs do not use parentheses.
2) On the java side, does `count_` not work with `agg_all_by`?  I'm trying to understand if the lack of support is because Java can't do it or if it is some weirdness in how we've done the python.